apps/yggdrasil/priv/repo/migrations/20160229163000_create_role_permission.exs

```diff
@@ -0,0 +1,16 @@
+defmodule Yggdrasil.Repo.Migrations.CreateRoleResource do
+  use Ecto.Migration
+
+  def change do
+    create table(:roles_permissions, primary_key: false) do
+      add :role_id, references(:roles, on_delete: :nothing)
+      add :resource, references(:resources, on_delete: :nothing, column: :name, type: :string)
+      add :permission, references(:permissions, on_delete: :nothing, column: :name, type: :string)
+
+      timestamps
+    end
+    create index(:roles_permissions, [:role_id])
```

Comment:
http://www.postgresql.org/docs/9.4/static/indexes-unique.html

If read that correctly we can drop the index, as the unique index covers and it's just duplicating the index.